src/aboutToRenderDecorator.ts

```diff
@@ -0,0 +1,38 @@
+import * as vscode from "vscode";
+import TabnineInlineCompletionItem from "./inlineSuggestions/tabnineInlineCompletionItem";
+import suggestionShown from "./binary/requests/suggestionShown";
+import CompletionOrigin from "./CompletionOrigin";
+
+type DecoratedFunction<R, T1 extends vscode.TextDocument, T2, T3, T4> = (
+  document: T1,
+  ...args: [T2, T3, T4]
+) => Promise<R>;
+
+export default function aboutToRenderDecorator<
```

Comment:
why like this? why not inline this functionality inside `provideInlineCompletionItems`?
this is much less readable